vm-virtio/src/vhost_user/net.rs

```diff
@@ -30,14 +30,9 @@ impl VhostUserMasterReqHandler for SlaveReqHandler {}
 
 pub struct Net {
     vhost_user_net: Master,
-    kill_evt: Option<EventFd>,
-    avail_features: u64,
-    acked_features: u64,
     backend_features: u64,
-    config_space: Vec<u8>,
     queue_sizes: Vec<u16>,
-    queue_evts: Option<Vec<EventFd>>,
-    interrupt_cb: Option<Arc<VirtioInterrupt>>,
+    net: VirtioNet,
```

Comment:
Referencing an object that implements the `VirtioDevice` trait inside a structure that also implements that trait is a little weird.
Although doing so does allow us to reduce code duplication, it also shows that we're doing something wrong and that both the vhost-user-net and virtio-net devices should use the same `VirtioDevice` implementation.

Reply:
Basically I think we should have a virtio backend trait, similar to what we're doing in the vhost_user_backend crate.
virtio-net and vhost-user-net should only have to provide an implementation for that backend. Then our VirtioDevice implementation for networking device would use that backend trait as a generic type.

Reply:
Yes I agree with all the statements here. This current factorization highlights a more global issue that we basically share everything between all VirtioDevice implementation. By having some kind of generic/intermediate trait, we would abstract and factorize a bunch of duplicated code.

Reply:
Ah, I got the point, let me work out a new version.